wcfsetup/install/files/ts/WoltLabSuite/Core/Media/Manager/Editor.ts

```diff
@@ -0,0 +1,403 @@
+      thumbnailOptions +=
+        '<option value="' +
+        thumbnailSize +
+        '">' +
+        Language.get("wcf.media.insert.imageSize." + thumbnailSize) +
+        "</option>";
+    });
+    thumbnailOptions += '<option value="original">' + Language.get("wcf.media.insert.imageSize.original") + "</option>";
+
+    const dialog = `
+      <div class="section">
+        <dl class="thumbnailSizeSelection">
+          <dt>${Language.get("wcf.media.insert.imageSize")}</dt>
+          <dd>
+            <select name="thumbnailSize">
+              ${thumbnailOptions}
+            </select>
+          </dd>
+        </dl>
+      </div>
+      <div class="formSubmit">
+        <button class="buttonPrimary">${Language.get("wcf.global.button.insert")}</button>
+      </div>`;
+
+    UiDialog.open({
+      _dialogSetup: () => {
+        return {
+          id: this._getInsertDialogId(),
+          options: {
+            onClose: this._editorClose.bind(this),
+            onSetup: (content) => {
+              content.querySelector(".buttonPrimary")!.addEventListener("click", (ev) => this._insertMedia(ev));
+
+              (content.querySelector(".thumbnailSizeSelection") as HTMLElement).style.display = "block";
+            },
+            title: Language.get("wcf.media.insert"),
+          },
+          source: dialog,
+        };
+      },
+    });
+  }
+
+  protected _click(event: MouseEvent): void {
+    this._activeButton = event.currentTarget;
+
+    super._click(event);
+  }
+
+  protected _dialogShow(): void {
+    super._dialogShow();
+
+    // check if data needs to be uploaded
+    if (this._uploadData) {
+      const fileUploadData = this._uploadData as OnDropPayload;
+      if (fileUploadData.file) {
+        this._upload.uploadFile(fileUploadData.file);
+      } else {
+        const blobUploadData = this._uploadData as PasteFromClipboard;
+        this._uploadId = this._upload.uploadBlob(blobUploadData.blob);
+      }
+
+      this._uploadData = null;
+    }
+  }
+
+  /**
+   * Handles pasting and dragging and dropping files into the editor.
+   */
+  protected _editorUpload(data: OnDropPayload): void {
+    this._uploadData = data;
+
+    UiDialog.open(this);
+  }
+
+  /**
+   * Returns the id of the insert dialog based on the media files to be inserted.
+   */
+  protected _getInsertDialogId(): string {
+    let dialogId = "mediaInsert";
+
+    this._mediaToInsert.forEach((media, mediaId) => {
+      dialogId += `-${mediaId}`;
+    });
+
+    return dialogId;
+  }
+
+  /**
+   * Returns the supported thumbnail sizes (excluding `original`) for all media images to be inserted.
+   */
+  protected _getThumbnailSizes(): string[] {
+    return ["small", "medium", "large"]
+      .map((size) => {
+        let supportSize = true;
+        this._mediaToInsert.forEach(function (media) {
+          if (!media[size + "ThumbnailType"]) {
+            supportSize = false;
+          }
+        });
```

Comment:
This can be `.every` when combined with `Array.from`. Similar code can be found below.